File: c#/bubble_sort.cs
public static int[] BubbleSort(int[] arr)
{
  bool noSwaps;
  for (int i = 0; i < arr.Length; i++)
  {
    noSwaps = true;
    for (int j = 0; j < arr.Length - 1; j++)
    {
      if (arr[j] > arr[j + 1])
      {
        int temp = arr[j + 1];
        arr[j + 1] = arr[j];
        arr[j] = temp;
        noSwaps = false;
      }
    }
    if (noSwaps)
    {
      break;
    }
  }

  return arr;
}

File: javascript/bubble_sort.js
function bubbleSort(arr) {
  for (let i = 0; i < arr.length; i++) {
    noSwaps = true
    for (let j = 0; j < arr.length; j++) {
      if (arr[j] > arr[j + 1]) {
        let temp = arr[j + 1]
        arr[j + 1] = arr[j]
        arr[j] = temp
        noSwaps = false
      }
    }
    if (noSwaps)
      break
  }
  return arr
}